mne/utils/tests/test_numerics.py

```diff
@@ -174,6 +174,32 @@ def test_time_mask():
     pytest.raises(ValueError, _time_mask, x[:1], tmin=10, sfreq=1)
 
 
+def test_freq_mask():
+    """Test safe frequency masking."""
+    N = 10
+    x = np.arange(N).astype(float)
+    assert _freq_mask(x, 0, N - 1, sfreq=1000.).sum() == N
+    assert _freq_mask(x - 1e-10, 0, N - 1, sfreq=1000.).sum() == N
+    assert _freq_mask(x - 1e-10, None, N - 1, sfreq=1000.).sum() == N
+    assert _freq_mask(x - 1e-10, None, None, sfreq=1000.).sum() == N
+    assert _freq_mask(x - 1e-10, -np.inf, None, sfreq=1000.).sum() == N
+    assert _freq_mask(x - 1e-10, None, np.inf, sfreq=1000.).sum() == N
+    # non-uniformly spaced inputs
+    x = np.array([4, 10])
+    assert _freq_mask(x[:1], fmin=10, sfreq=1, raise_error=False).sum() == 0
+    assert _freq_mask(x[:1], fmin=11, fmax=12, sfreq=1,
+                      raise_error=False).sum() == 0
+    assert _freq_mask(x, fmin=10, sfreq=1).sum() == 1
+    assert _freq_mask(x, fmin=6, sfreq=1).sum() == 1
+    assert _freq_mask(x, fmin=5, sfreq=1).sum() == 1
+    assert _freq_mask(x, fmin=4.5001, sfreq=1).sum() == 1
+    assert _freq_mask(x, fmin=4.4999, sfreq=1).sum() == 2
+    assert _freq_mask(x, fmin=4, sfreq=1).sum() == 2
+    # degenerate cases
+    pytest.raises(ValueError, _freq_mask, x[:1], fmin=11, fmax=12)
+    pytest.raises(ValueError, _freq_mask, x[:1], fmin=10, sfreq=1)
```

Comment:
FYI the modern way of testing this is more explicit and more targeted with:
```
with pytest.raises(ValueError, match='whatever substring regex is reasonable'):
    _freq_mask(...)
with pytest.raises(ValueError, match='whatever other substring regex is reasonable')
    _freq_mask(...)
```
But it's not worth holding up the PR for this, especially since it's already testing just a private function. As one of my professors used to say, perhaps "this one you do if you have an intrinsic motivator that kicks in" :)

However, it would be a worthwhile improvement to make some (all?) of these tests use the *public* API, namely in `mne/time_frequency/tests/test_tfr.py::test_crop` with e.g. `tfr_crop = tfr.copy().crop(...)`. As many of these private tests that can be adequately moved there should be. It both makes private function modifications easier in the future, and ensures that the public API is stable. The latter is ensured by the current changes, but indirectly rather than directly.

If it's not clear what I mean or if you do not want to/can't do this, I can push a quick commit for it.

Reply:
(In other words, it's better to explicitly ensure the public API is stable and only indirectly test the private functions, rather than explicitly test the private functions and only indirectly ensure the public API is stable.)

Reply:
Okay, do you mean: Scrap ```test_freq_mask``` / ```test_time_mask``` and make sure that the same functionality is instead tested by ```mne/time_frequency/tests/test_tfr.py::test_crop```?